### PR TITLE
0.2/673 onboarding tool medium

### PR DESCRIPTION
**Pull request by charlesstrange2**
Some commands for making things work

### CODE REVIEW

File: app/Console/Commands/AddCentre.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                $this->centre->sponsor()->associate($this->sponsor);
+                $this->centre->save();
+
+                exit("Done.\n");
+        }
+    }
+
+    public function warnUser()
+    {
+        $this->info('This unsafe command will alter the database to add records.');
```
Comment (katjam) [inline app/Console/Commands/AddCentre.php:125]: That sounds like we want them to say no...

How about: This action will add Centre with name, using code prefix xx to sponsor xx, with pref xx authorised by email.

File: app/Console/Commands/AddSponsor.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+                $this->sponsor->save();
+
+                exit("Done.\n");
+        }
+    }
+
+    public function warnUser()
+    {
+        $this->info('This unsafe command will alter the database to add records.');
```
Comment (katjam) [inline app/Console/Commands/AddSponsor.php:98]: same as centre - it's what the command is for. Confusing that we call it unsafe now.

File: app/Console/Commands/AddSponsor.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                    $this->sponsor->shortcode .
+                    " exists, exit without change.\n");
+
+            default:
+                // Check the user is happy to proceed
+                if (!$this->warnUser()) {
+                    exit("Exit without change.\n");
+                };
+                // Log the user in.
+                Auth::login($this->user);
```
Comment (katjam) [inline app/Console/Commands/AddSponsor.php:78]: Is it possible or desired to prompt for password - or is that taking it too far?

File: app/Console/Commands/RegisterFamiliesFromFile.php [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    /** @var RowCollection $rows */
+    private $rows;
+
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'arc:registerFamiliesFromFile
+                                {file : CSV file of family registrations to import, one per line}
```
Comment (katjam) [inline app/Console/Commands/RegisterFamiliesFromFile.php:36]: Scary! :)

File: app/Console/Commands/RegisterFamiliesFromFile.php [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+                $this->rows->each(function (CellCollection $row) use ($bar) {
+
+                    $signup_date = Carbon::createFromFormat('Y-m-d', $row->signup_date)->startOfDay();
+
+                    $registration = new Registration([
+                        'fm_chart_on' => ($row->pie_chart == "Y") ? $signup_date : null,
+                        'fm_diary_on' => ($row->pie_chart == "Y") ? $signup_date : null,
+                        'consented_on' => $signup_date,
+                        'eligibility' => 'other',
+                        ''
```
Comment (katjam) [inline app/Console/Commands/RegisterFamiliesFromFile.php]: What's this one for?

File: app/Console/Commands/RegisterFamiliesFromFile.php [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+                    $family->centre_sequence = (int) preg_replace(
+                        "/". $this->centre->prefix . "/",
+                        "",
+                        $row->seq_reference
+                    );
+
+                    // Make some carers
+                    $carer = $row->primary_carer_name;
+                    $carers = array_values(
+                        array_filter(
```
Comment (katjam) [inline app/Console/Commands/RegisterFamiliesFromFile.php:127]: Hard to follow by just reading code without seeing data too... assuming you have tested.

File: app/Console/Kernel.php [hunk 7]
```diff
@@ -4,6 +4,7 @@
 
 use Illuminate\Console\Scheduling\Schedule;
 use Illuminate\Foundation\Console\Kernel as ConsoleKernel;
+use Ramsey\Uuid\Generator\CombGenerator;
```
Comment (katjam) [inline app/Console/Kernel.php]: ooooh what does the comb generator do?

**Review by katjam — APPROVED** [review]
Looks ok in principle... just a few comments

File: app/Console/Kernel.php (hunk 7 above)
Comment (charlesstrange2) [reply app/Console/Kernel.php]: I dunno! IDE flags it a unused. I may have got helpfully added by a typo?

File: app/Console/Commands/RegisterFamiliesFromFile.php (hunk 6 above)
Comment (charlesstrange2) [reply app/Console/Commands/RegisterFamiliesFromFile.php:127]: I've run it through the real stuff they've said we can see. *seems* to work...

File: app/Console/Commands/RegisterFamiliesFromFile.php (hunk 5 above)
Comment (charlesstrange2) [reply app/Console/Commands/RegisterFamiliesFromFile.php]: No reason; removed.

File: app/Console/Commands/AddSponsor.php (hunk 3 above)
Comment (charlesstrange2) [reply app/Console/Commands/AddSponsor.php:78]: Too far i think; you need console access to do this, but at some point we might want to know whom was assigned responsibility for it (when we get round to finishing the lifecycle)

File: app/Console/Commands/AddCentre.php (hunk 1 above)
Comment (charlesstrange2) [reply app/Console/Commands/AddCentre.php:125]: It's a reminder to get your stuff together before you press y. perhaps redundant.

File: app/Console/Commands/AddSponsor.php (hunk 2 above)
Comment (charlesstrange2) [reply app/Console/Commands/AddSponsor.php:98]: It's a reminder to get your stuff together before you press y. perhaps redundant.

File: app/Console/Commands/AddCentre.php (hunk 1 above)
Comment (katjam) [reply app/Console/Commands/AddCentre.php:125]: Good idea to check - but I think more explicit would be useful rather than just a negative and blind - this is a dumb idea are you sure you want to go through with it.

File: app/Console/Commands/AddSponsor.php (hunk 3 above)
Comment (katjam) [reply app/Console/Commands/AddSponsor.php:78]: Agree

File: app/Console/Commands/RegisterFamiliesFromFile.php (hunk 6 above)
Comment (katjam) [reply app/Console/Commands/RegisterFamiliesFromFile.php:127]: Awesome

File: app/Console/Kernel.php (hunk 7 above)
Comment (katjam) [reply app/Console/Kernel.php]: lol
